pyccel/ast/numpyext.py

```diff
@@ -786,22 +786,22 @@ def __new__(cls, a, dtype=None, order='K', subok=True, shape=None):
 class NumpyNorm(PyccelInternalFunction):
     """ Represents call to numpy.norm"""
     _dtype = NativeReal()
+    _shape = ()
+    _rank  = 0
 
     def __init__(self, arg, dim=None):
-        if isinstance(dim, ValuedArgument):
-            dim = dim.value
-        if self.dim is not None:
-            sh = list(sh)
-            del sh[self.dim]
-            self._shape = tuple(sh)
-        else:
-            self._shape = ()
-        self._rank = len(self._shape)
         super().__init__(arg, dim)
+        if isinstance(arg.dtype, NativeBool):
+            arg = PythonFloat(PythonInt(arg))
```

Comment:
Why do you need to call 2 casting functions? Will `PythonFloat(arg)` not achieve the same result?

Reply:
I'm not sure of the relevance `PythonFloat(arg)` returns a `REAL` type

Reply:
by `REAL` type i mean in Fortran, in Pyccel there's NativeFloat

Reply:
Yes I understand, my question is why do you need `PythonInt`?

Reply:
i think there is three reasons for this:
1- PythonFloat doesn’t handle bool type
2- if we do `arg = PythonFloat(arg)` instead, it will be translated to something like 'real(TRUE)' which not acceptable in Fortran.
3- if have to handle bool type in PythonFloat i think we will do the same thing, cast `arg` to int type and then to float.

Reply:
It does now. I think it was fixed in PR #783

Reply:
See [here](https://github.com/pyccel/pyccel/blob/master/pyccel/codegen/printing/fcode.py#L786-L787)